lib/utils.js

```diff
@@ -95,11 +95,7 @@ utils.mcHexDigest = function mcHexDigest (hash, encoding) {
   // check for negative hashes
   const negative = hash.readInt8(0) < 0
   if (negative) performTwosCompliment(hash)
-  let digest = hash.toString('hex')
-  // trim leading zeroes
-  digest = digest.replace(/^0+/g, '')
-  if (negative) digest = '-' + digest
-  return digest
+  return negative ? '-' : '' + hash.toString('hex').replace(/^0+/g, '')
```

Comment:
You've created a monster there. add `(` and `)` around the terenary. Right now its `-` for negative and hash for !negative